src/python/pants/jvm/resolve/coursier_setup.py

```diff
@@ -109,22 +109,24 @@ class CoursierSubsystem(TemplatedExternalTool):
     name = "coursier"
     help = "A dependency resolver for the Maven ecosystem. (https://get-coursier.io/)"
 
-    default_version = "v2.1.0-M5-18-gfebf9838c"
+    default_version = "v2.1.6"
     default_known_versions = [
+        "v2.1.6|macos_arm64 |746b3e346fa2c0107fdbc8a627890d495cb09dee4f8dcc87146bdb45941088cf|20829782|https://github.com/VirtusLab/coursier-m1/releases/download/v2.1.6/cs-aarch64-apple-darwin.gz",
+        "v2.1.6|linux_arm64 |33330ca433781c9db9458e15d2d32e5d795de3437771647e26835e8b1391af82|20899290|https://github.com/VirtusLab/coursier-m1/releases/download/v2.1.6/cs-aarch64-pc-linux.gz",
+        "v2.1.6|linux_x86_64|af7234f8802107f5e1130307ef8a5cc90262d392f16ddff7dce27a4ed0ddd292|20681688",
+        "v2.1.6|macos_x86_64|36a5d42a0724be2ac39d0ebd8869b985e3d58ceb121bc60389ee2d6d7408dd56|20037412",
         "v2.1.0-M5-18-gfebf9838c|linux_arm64 |d4ad15ba711228041ad8a46d848c83c8fbc421d7b01c415d8022074dd609760f|19264005",
         "v2.1.0-M5-18-gfebf9838c|linux_x86_64|3e1a1ad1010d5582e9e43c5a26b273b0147baee5ebd27d3ac1ab61964041c90b|19551533",
-        "v2.1.0-M5-18-gfebf9838c|macos_arm64 |d13812c5a5ef4c9b3e25cc046d18addd09bacd149f95b20a14e4d2a73e358ecf|18826510",
         "v2.1.0-M5-18-gfebf9838c|macos_x86_64|d13812c5a5ef4c9b3e25cc046d18addd09bacd149f95b20a14e4d2a73e358ecf|18826510",
         "v2.0.16-169-g194ebc55c|linux_arm64 |da38c97d55967505b8454c20a90370c518044829398b9bce8b637d194d79abb3|18114472",
         "v2.0.16-169-g194ebc55c|linux_x86_64|4c61a634c4bd2773b4543fe0fc32210afd343692891121cddb447204b48672e8|18486946",
-        "v2.0.16-169-g194ebc55c|macos_arm64 |15bce235d223ef1d022da30b67b4c64e9228d236b876c834b64e029bbe824c6f|17957182",
         "v2.0.16-169-g194ebc55c|macos_x86_64|15bce235d223ef1d022da30b67b4c64e9228d236b876c834b64e029bbe824c6f|17957182",
     ]
     default_url_template = (
         "https://github.com/coursier/coursier/releases/download/{version}/cs-{platform}.gz"
     )
     default_url_platform_mapping = {
-        "macos_arm64": "x86_64-apple-darwin",
+        "macos_arm64": "aarch64-apple-darwin",
```

Comment:
Relating to https://github.com/pantsbuild/pants/pull/20118 :
what has this done for support on M1 for different kinds of users?

An M1 user using pants-defaults for coursier will be fine whether or not they use Rosetta. The default version is bumped up to 2.1.6 and we provide a working version for it.

An M1 user who had pinned their coursier version in pants.toml will be broken assuming they were using Rosetta prior. The known versions are gone. Unfortunately, we cannot just restore the deleted known version lines for them without some code changes because `default_url_platform_mapping` is doing double duty:

- it determines what url to download, which we can override with the optional 5th field in known-versions
- it also determines the name of the file within the archive to extract as the `exe` because `self.generate_exe(plat)` for Coursier depends on the result of `self.generate_url(plat)`

so we cannot have it extract the exe named `...aarch64...` from the virtuslabs downloads in one case but download the `...x86_64....` from the regular urls in the other


We have a few options:
- Revert this PR and start over, maybe point users to Rosetta, which I found in local testing worked fine.
- Leave as is and hope that users that pinned the dep can unpin or bump it
- fix-forward and rework the interactions of the 5th field url-override, generate_exe and generate_url so that old pinned rosetta users are fine but unpinned users can use arm-compiled binaries.

Reply:
Hmm, maybe a combination of 1 and 3. Unless we can get 3 going soon, it may block 2.18.0 release, so 1 to unblock that and 3 for long term?

Reply:
+1 adding the revert to #20118